IllegalStateException: Fragment has not been attached yet

Crashlytics reports this crash has been happening all weekend every since the 0.23.1 supplemental release: https://fabric.io/automattic/android/apps/com.woocommerce.android/issues/5c302b15f8b88c29633bbf93?time=last-ninety-days

```
Fatal Exception: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.woocommerce.android/com.woocommerce.android.ui.main.MainActivity}: java.lang.IllegalStateException: Fragment has not been attached yet.
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2804)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2875)
       at android.app.ActivityThread.-wrap12(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1578)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:156)
       at android.app.ActivityThread.main(ActivityThread.java:6623)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:942)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:832)

Caused by java.lang.IllegalStateException: Fragment has not been attached yet.
       at android.support.v4.app.Fragment.instantiateChildFragmentManager(Fragment.java:2300)
       at android.support.v4.app.Fragment.getChildFragmentManager(Fragment.java:763)
       at com.woocommerce.android.ui.base.TopLevelFragment.loadChildFragment(TopLevelFragment.kt:86)
       at com.woocommerce.android.ui.base.TopLevelFragmentView$DefaultImpls.openOrderDetail(TopLevelFragmentView.kt:98)
       at com.woocommerce.android.ui.base.TopLevelFragment.openOrderDetail(TopLevelFragment.kt:18)
       at com.woocommerce.android.ui.main.MainActivity.showNotificationDetail(MainActivity.kt:486)
       at com.woocommerce.android.ui.main.MainActivity.initFragment(MainActivity.kt:359)
       at com.woocommerce.android.ui.main.MainActivity.onCreate(MainActivity.kt:122)
       at android.app.Activity.performCreate(Activity.java:6910)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1123)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2757)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2875)
       at android.app.ActivityThread.-wrap12(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1578)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:156)
       at android.app.ActivityThread.main(ActivityThread.java:6623)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:942)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:832)
```

Update: I was just able to repro this by sending the app to the background, then tapping a review push notification.

```
Caused by: java.lang.IllegalStateException: Fragment has not been attached yet.
        at android.support.v4.app.Fragment.instantiateChildFragmentManager(Fragment.java:2383)
        at android.support.v4.app.Fragment.getChildFragmentManager(Fragment.java:845)
        at com.woocommerce.android.ui.base.TopLevelFragment.getFragmentFromBackStack(TopLevelFragment.kt:79)
        at com.woocommerce.android.ui.notifications.NotifsListFragment.openReviewDetail(NotifsListFragment.kt:301)
        at com.woocommerce.android.ui.main.MainActivity.showNotificationDetail(MainActivity.kt:396)
        at com.woocommerce.android.ui.main.MainActivity.initFragment(MainActivity.kt:357)
        at com.woocommerce.android.ui.main.MainActivity.onNewIntent(MainActivity.kt:137)
        at android.app.Activity.performNewIntent(Activity.java:7149)
```

--- a/RELEASE-NOTES.txt
+++ b/RELEASE-NOTES.txt
@@ -19,6 +19,7 @@ Bugfixes
 - Fixed a crash that occasionally happened while attempting to load the app from a woocommerce notification alert
 - Fixed a rare crash when showing the Google login screen
 - Fixed bug that could cause the back arrow to appear in the toolbar when it shouldn't
+- Fixed crash when opening app from a notification
 
 Improvements
 - Custom order status labels are now supported! Instead of just displaying the order status slug, the custom order

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/base/TopLevelFragment.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/base/TopLevelFragment.kt
@@ -76,7 +76,11 @@ abstract class TopLevelFragment : Fragment(), TopLevelFragmentView {
     }
 
     override fun getFragmentFromBackStack(tag: String): Fragment? {
-        return childFragmentManager.findFragmentByTag(tag)
+        return if (isAdded) {
+            childFragmentManager.findFragmentByTag(tag)
+        } else {
+            null
+        }
     }
 
     override fun popToState(tag: String): Boolean {

